subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/tasks/JacocoMerge.java

```diff
@@ -86,6 +91,7 @@ protected IsolatedAntBuilder getAntBuilder() {
 
     @TaskAction
     public void merge() {
+        DeprecationLogger.deprecateTaskType(getClass(), getPath()).replaceWith(JacocoReport.class).willBeRemovedInGradle8().withUpgradeGuideSection(7, "jacoco_merge").nagUser();
```

Comment:
Is there a reason for using `getClass()` instead of `JacocoMerge.class` here?
The latter would not require the `"_Decorated"` check in `DeprecateTaskType.deprecateTaskType`.

Reply:
There's no particular reason. I'll adjust the PR.

Reply:
Done.